From 54333280d2265d3d83e69f6c5e1f3656f85f2ade Mon Sep 17 00:00:00 2001
From: Robert Lehmann <robert.lehmann@sitec-systems.de>
Date: Tue, 1 Nov 2016 08:09:29 +0000
Subject: [PATCH 1018/1018] mach-imx6sx: Add fixup for ksz8091 phy

The fixup will enable the LED mode of the phy and sets it into the
second LED mode of the device. LED0 Shows if a link is detected and
LED1 shows the Activity on the network.

Ticket #1470
---
 arch/arm/mach-imx/mach-imx6sx.c | 27 +++++++++++++++++++++++++--
 1 file changed, 25 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-imx/mach-imx6sx.c b/arch/arm/mach-imx/mach-imx6sx.c
index 0d72478..b5dc376 100644
--- a/arch/arm/mach-imx/mach-imx6sx.c
+++ b/arch/arm/mach-imx/mach-imx6sx.c
@@ -15,6 +15,7 @@
 #include <linux/gpio.h>
 #include <asm/mach/arch.h>
 #include <asm/mach/map.h>
+#include <linux/micrel_phy.h>
 
 #include "common.h"
 #include "hardware.h"
@@ -46,14 +47,36 @@ static int ar8031_phy_fixup(struct phy_device *dev)
 	return 0;
 }
 
+static int ksz8091_phy_fixup(struct phy_device *dev)
+{
+	u16 val;
+
+	// Disable PME and enable LED0
+	val = phy_read(dev, 0x16);
+	val &= ~(0x8000);
+	phy_write(dev, 0x16, val);
+	
+	// Setting LED mode to Mode 01
+	val = phy_read(dev, 0x1F);
+	val |= 0x0010;
+	phy_write(dev, 0x1F, val);
+	
+	return 0;
+}
+
 #define PHY_ID_AR8031   0x004dd074
 static void __init imx6sx_enet_phy_init(void)
 {
-	if (IS_BUILTIN(CONFIG_PHYLIB))
+	if (IS_BUILTIN(CONFIG_PHYLIB)) {
 		phy_register_fixup_for_uid(PHY_ID_AR8031, 0xffffffff,
-					   ar8031_phy_fixup);
+								   ar8031_phy_fixup);
+		phy_register_fixup_for_uid(PHY_ID_KSZ8081, MICREL_PHY_ID_MASK,
+								   ksz8091_phy_fixup);
+	}
+	
 }
 
+
 static void __init imx6sx_enet_clk_sel(void)
 {
 	struct regmap *gpr;
-- 
2.1.4

